File: 2606-바이러스/main.cpp
#include <iostream>
#include <vector>
using namespace std;

int virus = 0;

void dfs(vector<vector<int>>& web, vector<bool>& visited, int start) {
  virus++;
  visited[start] = true;

  for (auto c : web[start]) {
    if(!visited[c])
      dfs(web, visited, c);
  }
}

int main() {
  int N, T, from, to;
  cin >> N;
  cin >> T;

  vector<vector<int>> web(N + 1);
  vector<bool> visited(N + 1, false);

  for (int i = 0; i < T; i++) {
    cin >> from >> to;
    web[from].push_back(to);
    web[to].push_back(from);
  }

  dfs(web, visited, 1);
  cout << virus - 1;
}

File: 2606-바이러스/main.py
import sys
input = sys.stdin.readline

virus = 0

def dfs(web: list[list[int]], visited: list[bool], start: int):
  global virus
  virus += 1
  visited[start] = True

  for c in web[start]:
    if not visited[c]:
      dfs(web, visited, c)

N = int(input())
T = int(input())

web = [[] for _ in range(N + 1)]
visited = [False for _ in range(N + 1)]

for i in range(T):
  X, Y = map(int, input().split())
  web[X].append(Y)
  web[Y].append(X)

dfs(web, visited, 1)
print(virus - 1)